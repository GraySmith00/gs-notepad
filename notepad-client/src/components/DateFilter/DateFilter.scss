.date-filter {
  margin-right: 2rem;
}
