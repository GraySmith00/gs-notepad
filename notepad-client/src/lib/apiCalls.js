export const postNote = async note => {
  const url = process.env.REACT_APP_POST_NOTE;
  const response = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(note),
    headers: {
      'Content-Type': 'application/json'
    }
  });

  if (response.status > 299) {
    const error = await response.json();
    throw new Error(error.message);
  }

  const data = await response.json();
  return data;
};

export const getNotes = async () => {
  const url = process.env.REACT_APP_GET_NOTES;
  const response = await fetch(url);
  const data = await response.json();
  return data;
};
