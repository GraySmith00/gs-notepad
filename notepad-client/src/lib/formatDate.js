import moment from 'moment';

export const formatDate = timeStamp => {
  return moment(timeStamp).format('L');
};
