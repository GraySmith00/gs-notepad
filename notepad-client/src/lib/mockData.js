export const mockNotes = [
  {
    createdAt: 1545951483012,
    text: 'buy groceries',
    tag: 'hobby',
    id: 'dd5aa440-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951433152,
    text: 'do laundry',
    tag: 'work',
    id: 'bfa29c00-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951460092,
    text: 'play golf',
    tag: 'hobby',
    id: 'cfb153c0-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951504338,
    text: 'buy gifts',
    tag: 'personal',
    id: 'ea10bb20-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545451632475,
    text: 'set up tv',
    tag: 'personal',
    id: '425ca4b0-0a2b-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951440518,
    text: 'run errands',
    tag: 'work',
    id: 'c4069260-0a2a-11e9-93fa-a7949507e226'
  }
];
