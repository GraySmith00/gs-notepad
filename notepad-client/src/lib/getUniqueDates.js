import { formatDate } from './formatDate';

export const getUniqueDates = notes => {
  const dates = notes.reduce((dates, note) => {
    const formattedDate = formatDate(note.createdAt);

    if (!dates[formattedDate]) {
      dates[formattedDate] = 1;
    }

    return dates;
  }, {});

  return Object.keys(dates);
};
