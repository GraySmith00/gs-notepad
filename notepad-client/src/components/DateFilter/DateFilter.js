import React from 'react';
import PropTypes from 'prop-types';
import { getUniqueDates } from '../../lib/getUniqueDates';

const DateFilter = ({ setDateFilter, notes }) => {
  return (
    <form className="date-filter">
      <select onChange={e => setDateFilter(e.target.value)}>
        <option value="all">All Dates</option>
        {getUniqueDates(notes).map(date => (
          <option key={date} value={date}>
            {date}
          </option>
        ))}
      </select>
    </form>
  );
};

DateFilter.propTypes = {
  setDateFilter: PropTypes.func.isRequired,
  notes: PropTypes.array.isRequired
};

export default DateFilter;
