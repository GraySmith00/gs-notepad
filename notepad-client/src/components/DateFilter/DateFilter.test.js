import React from 'react';
import { shallow } from 'enzyme';
import DateFilter from './DateFilter';
import { mockNotes } from '../../lib/mockData';

describe('DateFilter component', () => {
  let wrapper;
  let mockSetDateFilter;

  beforeEach(() => {
    mockSetDateFilter = jest.fn();
    wrapper = shallow(
      <DateFilter notes={mockNotes} setDateFilter={mockSetDateFilter} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
