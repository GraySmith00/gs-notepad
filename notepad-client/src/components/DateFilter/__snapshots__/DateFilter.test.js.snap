// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateFilter component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DateFilter
    notes={
      Array [
        Object {
          "createdAt": 1545951483012,
          "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
          "tag": "hobby",
          "text": "buy groceries",
        },
        Object {
          "createdAt": 1545951433152,
          "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
          "tag": "work",
          "text": "do laundry",
        },
        Object {
          "createdAt": 1545951460092,
          "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
          "tag": "hobby",
          "text": "play golf",
        },
        Object {
          "createdAt": 1545951504338,
          "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
          "tag": "personal",
          "text": "buy gifts",
        },
        Object {
          "createdAt": 1545451632475,
          "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
          "tag": "personal",
          "text": "set up tv",
        },
        Object {
          "createdAt": 1545951440518,
          "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
          "tag": "work",
          "text": "run errands",
        },
      ]
    }
    setDateFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[Function]}
      >
        <option
          value="all"
        >
          All Dates
        </option>
        <option
          value="12/27/2018"
        >
          12/27/2018
        </option>
        <option
          value="12/21/2018"
        >
          12/21/2018
        </option>
      </select>,
      "className": "date-filter",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="all"
          >
            All Dates
          </option>,
          Array [
            <option
              value="12/27/2018"
            >
              12/27/2018
            </option>,
            <option
              value="12/21/2018"
            >
              12/21/2018
            </option>,
          ],
        ],
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All Dates",
            "value": "all",
          },
          "ref": null,
          "rendered": "All Dates",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "12/27/2018",
          "nodeType": "host",
          "props": Object {
            "children": "12/27/2018",
            "value": "12/27/2018",
          },
          "ref": null,
          "rendered": "12/27/2018",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "12/21/2018",
          "nodeType": "host",
          "props": Object {
            "children": "12/21/2018",
            "value": "12/21/2018",
          },
          "ref": null,
          "rendered": "12/21/2018",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[Function]}
        >
          <option
            value="all"
          >
            All Dates
          </option>
          <option
            value="12/27/2018"
          >
            12/27/2018
          </option>
          <option
            value="12/21/2018"
          >
            12/21/2018
          </option>
        </select>,
        "className": "date-filter",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="all"
            >
              All Dates
            </option>,
            Array [
              <option
                value="12/27/2018"
              >
                12/27/2018
              </option>,
              <option
                value="12/21/2018"
              >
                12/21/2018
              </option>,
            ],
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All Dates",
              "value": "all",
            },
            "ref": null,
            "rendered": "All Dates",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "12/27/2018",
            "nodeType": "host",
            "props": Object {
              "children": "12/27/2018",
              "value": "12/27/2018",
            },
            "ref": null,
            "rendered": "12/27/2018",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "12/21/2018",
            "nodeType": "host",
            "props": Object {
              "children": "12/21/2018",
              "value": "12/21/2018",
            },
            "ref": null,
            "rendered": "12/21/2018",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
